migrations/2024-12-17-144959_community-post-tags/up.sql

```diff
@@ -6,7 +6,9 @@
 CREATE TABLE tag (
     id serial PRIMARY KEY,
     ap_id text NOT NULL UNIQUE,
-    display_name text NOT NULL,
+    name varchar(255) NOT NULL,
+    description text,
+    background_color varchar(7),
```

Comment:
This I'm against. No apps or APIs should not be hardcoding colors nowadays, as its not going to work with different themes, dark/light modes, and platforms. 

Android material, bootstrap, and other frameworks have the right idea, where you only define colors as Primary, Secondary, and Tertiary as enums, and let the theme itself pick from its own list of colors. If you want to keep this, I'd suggest just making it a simple int, that we could potentially turn into an enum later. Then at least apps could use that simple number to pick their own theme color.

Reply:
Instead of storing a specific color, store a `index_within_community int NOT NULL` column that auto-increments separately for each community. In a `BEFORE INSERT` trigger, the next value can be generated using `UPDATE community SET last_tag_index = last_tag_index + 1 WHERE id = NEW.community_id RETURNING last_tag_index`.

Lemmy-ui can assign a color to each `index_within_community` value by using the hues for red and blue for the first 2 tags, then using a binary van der Corput sequence for the hues in between. For example, the first 5 tags' hues would be:
1. 0
2. 240
3. 120 (between 0 and 240)
4. 60 (between 0 and 120)
5. 180 (between 120 and 240)

https://en.wikipedia.org/wiki/Van_der_Corput_sequence

Reply:
Cool, made #5880 to track this and discuss there.